### PR TITLE
Add log system support for enclave

### PR DESCRIPTION
**Pull request by whh39**
Add log system support for enclave

test: passed all the unit test
Signed-off-by: wanghaiheng <haihengx.wang@intel.com>

### CODE REVIEW

File: utils/log4cplus/ulog_utils.h [hunk 1]
```diff
@@ -42,38 +42,62 @@ using namespace log4cplus::helpers;
...
+                                      MAX_LOG_BUF,                \
+                                      pFormat,                    \
+                                      ##__VA_ARGS__)
+#define log_w(pFormat, ...) do_logger(log4cplus::WARN_LOG_LEVEL, \
+                                      __FILE__,                  \
+                                      __LINE__,                  \
+                                      MAX_LOG_BUF,               \
+                                      pFormat,                   \
+                                      ##__VA_ARGS__)
+#define elog_d(pFormat, x, y, ...) do_logger(log4cplus::DEBUG_LOG_LEVEL, \
```
Comment (wanghouqi) [inline utils/log4cplus/ulog_utils.h:74]: x, y 改为file, line

Comment (syan10) [reply utils/log4cplus/ulog_utils.h:74]: elog_x 只是给enclave里用的吗， 为啥不定义再elog_utils.h里面

Comment (wanghouqi) [reply utils/log4cplus/ulog_utils.h:74]: 将elog_d,elog_i,elog_e,elog_w,合并为log_c(pFormat, logLevel, filename, line, ...) //print custom log

Comment (whh39) [reply utils/log4cplus/ulog_utils.h]: 改好了

**Review by syan10 — APPROVED** [review]

**Pull request merged**